pytorch_examples/ex_utils.py

```diff
@@ -0,0 +1,58 @@
+import png
+from PIL import Image
+import numpy as np
+from scipy.ndimage import gaussian_filter
+
+# Wrapper for reading images
+def imread(filename, bits=8):
+    assert bits in (8, 16, 32), 'Unsupported bit-depth'
+    if filename.endswith('.png'):
+        pngdata = png.Reader(filename).read_flat()
+        img = np.array(pngdata[2]).reshape((pngdata[1], pngdata[0], -1))
+    else:
+        img = np.array(Image.open(filename).convert("RGB"))
+    return img
+
+
+# Add 0 mean Gaussian noise
+def imnoise(clean, std, static=False):
+    dtype = clean.dtype
```

Comment:
You could use https://numpy.org/doc/stable/reference/generated/numpy.iinfo.html instead of the "if" statements.